circuit-construction/src/prover.rs

```diff
@@ -12,94 +12,9 @@ use kimchi::{
     proof::ProverProof,
     prover_index::ProverIndex,
 };
-use mina_curves::pasta::{Fp, Fq, Pallas as Other, Vesta};
 use oracle::FqSponge;
 use std::array;
 
-/// A [Cycle] represents the algebraic structure that
-/// allows for recursion using elliptic curves.
-pub trait Cycle {
-    type InnerField: FftField
-        + PrimeField
-        + SquareRootField
-        + From<u128>
-        + From<u64>
-        + From<u32>
-        + From<u16>
-        + From<u8>;
-    type OuterField: FftField
-        + PrimeField
-        + SquareRootField
-        + From<u128>
-        + From<u64>
-        + From<u32>
-        + From<u16>
-        + From<u8>;
-
-    type InnerMap: groupmap::GroupMap<Self::InnerField>;
-    type OuterMap: groupmap::GroupMap<Self::OuterField>;
-
-    type InnerProj: ProjectiveCurve<
-            Affine = Self::Inner,
-            ScalarField = Self::OuterField,
-            BaseField = Self::InnerField,
-        > + From<Self::Inner>
-        + Into<Self::Inner>
-        + std::ops::MulAssign<Self::OuterField>;
-
-    type Inner: CommitmentCurve<
-            Projective = Self::InnerProj,
-            Map = Self::InnerMap,
-            BaseField = Self::InnerField,
-            ScalarField = Self::OuterField,
-        > + From<Self::InnerProj>
-        + Into<Self::InnerProj>;
-
-    type OuterProj: ProjectiveCurve<
-            Affine = Self::Outer,
-            ScalarField = Self::InnerField,
-            BaseField = Self::OuterField,
-        > + From<Self::Outer>
-        + Into<Self::Outer>
-        + std::ops::MulAssign<Self::InnerField>;
-
-    type Outer: CommitmentCurve<
-            Projective = Self::OuterProj,
-            Map = Self::OuterMap,
-            ScalarField = Self::InnerField,
-            BaseField = Self::OuterField,
-        > + KimchiCurve;
-}
-
-/// Used to configure the base curve of Pallas
-pub struct FpInner;
-/// Used to configure the base curve of Vesta
-pub struct FqInner;
-
-impl Cycle for FpInner {
-    type InnerMap = <Other as CommitmentCurve>::Map;
-    type OuterMap = <Vesta as CommitmentCurve>::Map;
-
-    type InnerField = Fp;
-    type OuterField = Fq;
-    type Inner = Other;
-    type Outer = Vesta;
-    type InnerProj = <Other as AffineCurve>::Projective;
-    type OuterProj = <Vesta as AffineCurve>::Projective;
-}
-
-impl Cycle for FqInner {
-    type InnerMap = <Vesta as CommitmentCurve>::Map;
-    type OuterMap = <Other as CommitmentCurve>::Map;
-
-    type InnerField = Fq;
-    type OuterField = Fp;
-    type Inner = Vesta;
-    type Outer = Other;
-    type InnerProj = <Vesta as AffineCurve>::Projective;
-    type OuterProj = <Other as AffineCurve>::Projective;
-}
-
```

Comment:
All of this was replaced by `KimchiCurve`, also we should create a PR to deprecate this